### PR TITLE
libfuse: support team favorites and expose team list in dev mode

### PR DESCRIPTION
**Pull request by strib**
This vendors in keybase/client#7184, and supports a "team" top-level folderlist showing team favorites.  (Only in "devel" runmode for now.)

It also auto-favorites team folders in the local daemon (this is eventually what the API server should do as well).

With these changes, it's possible to access team folders in local mode, and when #1007 they work properly across restarts (while changing users, etc).

Issue: KBFS-2215

### CODE REVIEW

File: libfuse/dir.go [hunk 1]
```diff
@@ -349,6 +349,17 @@ func (f *Folder) isWriter(ctx context.Context) (bool, error) {
 	}
 	f.handleMu.RLock()
 	defer f.handleMu.RUnlock()
+	if f.h.Type() == tlf.SingleTeam {
+		tid, err := f.h.FirstResolvedWriter().AsTeam()
+		if err != nil {
+			return false, err
+		}
+		isWriter, err := f.fs.config.KBPKI().IsTeamWriter(ctx, tid, session.UID)
```
Comment (songgao) [inline libfuse/dir.go]: It seems this will be called pretty often. Do we need a short TTL cache in `libkbfs`?

Comment (strib) [inline libfuse/dir.go]: I'm planning to cache team membership stuff within `keybase_service_base`, so these calls won't be real RPCs most of the time.  Does that address your concern?

Comment (songgao) [inline libfuse/dir.go]: Yes, perfectly!

File: libkbfs/keybase_daemon.go [hunk 2]
```diff
@@ -59,13 +60,44 @@ func (k keybaseDaemon) NewKeybaseService(config Config, params InitParams, ctx C
 	localUID := localUsers[userIndex].UID
 	codec := config.Codec()
 
+	teams := MakeLocalTeams([]libkb.NormalizedUsername{"kbfs", "core", "dokan"})
+	for i := range teams {
+		teams[i].Writers = make(map[keybase1.UID]bool)
+		teams[i].Readers = make(map[keybase1.UID]bool)
+		switch teams[i].Name {
+		case "kbfs":
+			teams[i].Writers[localUsers[0].UID] = true // strib
```
Comment (jzila) [inline libkbfs/keybase_daemon.go]: Where are these defined?

Comment (strib) [inline libkbfs/keybase_daemon.go]: Just above here in the same function.

Comment (jzila) [inline libkbfs/keybase_daemon.go]: Hah, I guess all I had to do was click.

**Pull request merged**